### PR TITLE
Allow redis to be configured with TLS

### PR DESCRIPTION
**Pull request by MMartyn**

### CODE REVIEW

**Comment by MMartyn** [comment]
I opted to let the redis client parse the pieces so that I could optionally include a password in the host env var, but I could also switch this to having another env var for redis password if that would be preferred.

https://github.com/redis/redis-rb/blob/master/lib/redis/client.rb#L403

File: core/lib/workarea/configuration/redis.rb [hunk 1]
```diff
@@ -67,8 +71,14 @@ def db
         @config[:db]
       end
 
+      def to_h
```
Comment (bencrouse) [inline core/lib/workarea/configuration/redis.rb]: I'm curious, why switch this to be only a URL? How would I get a hash-version of the config now?

Comment (bencrouse) [reply core/lib/workarea/configuration/redis.rb]: Ah, I missed your comment above. I think for compatibility it'd be best to keep the hash version around, can you expand the hash to include all the things?

**Comment by bencrouse** [comment]
@MMartyn hey if you want to get this into v3.5, you'll need to update this week.

File: core/lib/workarea/configuration/redis.rb (hunk 1 above)
Comment (MMartyn) [reply core/lib/workarea/configuration/redis.rb]: Updated

File: core/lib/workarea/configuration/redis.rb [hunk 2]
```diff
@@ -39,11 +39,15 @@ def find_config(name)
...
+          scheme = ENV["WORKAREA_#{env_slug}_SCHEME"].presence || DEFAULT[:scheme]
 
           {
+            scheme: scheme,
             host: ENV["WORKAREA_#{env_slug}_HOST"].presence || DEFAULT[:host],
             port: ENV["WORKAREA_#{env_slug}_PORT"].presence || DEFAULT[:port],
-            db: ENV["WORKAREA_#{env_slug}_DB"].presence || DEFAULT[:db]
+            db: ENV["WORKAREA_#{env_slug}_DB"].presence || DEFAULT[:db],
+            password: ENV["WORKAREA_#{env_slug}_PASSWORD"].presence,
+            ssl: scheme == 'rediss' ? true : false
```
Comment (bencrouse) [inline core/lib/workarea/configuration/redis.rb]:
```suggestion
            ssl: scheme == 'redis' ? true : false
```

Comment (bencrouse) [reply core/lib/workarea/configuration/redis.rb]: @MMartyn I can't commit this suggestion since its your personal fork, can you commit that?

Comment (MMartyn) [reply core/lib/workarea/configuration/redis.rb]: I can see how it might seem like a typo, but rediss was intentional.

**Pull request merged**